tests/mr/device/cuda_async_mr_tests.cpp

```diff
@@ -24,7 +24,22 @@ namespace {
 
 using cuda_async_mr = rmm::mr::cuda_async_memory_resource;
 
-TEST(AsyncMRTest, ThrowIfNotSupported)
+class AsyncMRTest : public ::testing::Test {
+ protected:
+  void SetUp() override
+  {
+    int runtime_version{};
+    RMM_CUDA_TRY(cudaRuntimeGetVersion(&runtime_version));
+    int driver_version{};
+    RMM_CUDA_TRY(cudaDriverGetVersion(&driver_version));
```

Comment:
```suggestion
    int driver_supports_pool;
    RMM_CUDA_TRY(cudaDeviceGetAttribute(&driver_supports_pool,  cudaDevAttrMemoryPoolsSupported);
```

Reply:
This is also needed here: https://github.com/rapidsai/rmm/blob/d8dc71501657e820078bc26843565990d8052533/tests/mr/device/mr_test.hpp#L258-L261

We should wrap this logic in a function like `is_cuda_pool_available`.

Reply:
Maybe `static bool cuda_async_memory_resource::is_supported()`. Then use this in the `cuda_async_mr` ctor as well as here in the tests.

Reply:
I get a symbol not found error if I use `cudaDevAttrMemoryPoolsSupported` with cuda 11.0.

Reply:
Well you still have to guard it just like in the ctor using the flag we already have.  The problem we are solving isn't compiling with cuda 11.0. It's compiling with 11.2+ and running on <11.2.

Reply:
Done.